python/mxnet/gluon/model_zoo/vision/densenet.py

```diff
@@ -131,12 +132,16 @@ def get_densenet(num_layers, pretrained=False, ctx=cpu(), **kwargs):
         Whether to load the pretrained weights for model.
     ctx : Context, default CPU
         The context in which to load the pretrained weights.
+    root : str, default '~/.mxnet/models'
+        Location for keeping the model parameters.
     """
     num_init_features, growth_rate, block_config = densenet_spec[num_layers]
-    net = DenseNet(num_init_features, growth_rate, block_config, **kwargs)
+    net_args = _get_arg_dict(kwargs, ('bn_size', 'dropout', 'classes', 'prefix', 'params'))
```

Comment:
why use _get_arg_dict? Shouldn't extra arguments cause an error?

Reply:
kwargs include both arguments for the network and for the model zoo, feeding one to the other can cause unintended error.

Reply:
why not put model_zoo params explicitly in the interface?
like `get_densenet(num_layers, pretrained=False, ctx=cpu(), root=xxx, **kwarg`
The current code ignores extra arguments without warning

Reply:
Sure, this is doable. If you have strong preference on this I'm happy to make the change.